src/main/java/org/jboss/snowdrop/licenses/sanitiser/ExternalLicenseSanitiser.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016-2017 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.snowdrop.licenses.sanitiser;
+
+import org.jboss.snowdrop.licenses.provider.ExternalLicenseProvider;
+import org.jboss.snowdrop.licenses.xml.DependencyElement;
+
+import java.util.Optional;
+
+/**
+ * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ * <br>
+ * Date: 11/15/17
+ */
+public class ExternalLicenseSanitiser implements LicenseSanitiser {
+
+    private final LicenseSanitiser next;
+    private final Optional<ExternalLicenseProvider> provider;
+
+    public ExternalLicenseSanitiser(Optional<String> licenseServerUrl, LicenseSanitiser next) {
+        this.next = next;
+        this.provider = licenseServerUrl.map(ExternalLicenseProvider::new);
+    }
+    @Override
+    public DependencyElement fix(DependencyElement dependencyElement) {
+        return provider.flatMap(
+                p -> p.getLicenses(dependencyElement.toGav())
```

Comment:
If `provider` doesn't recognise the gav (i.e. returns an empty set), next sanitiser should be called.

Reply:
Done